homeassistant/components/elmax/cover.py

```diff
@@ -0,0 +1,137 @@
+"""Elmax cover platform."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from elmax_api.model.command import CoverCommand
+from elmax_api.model.cover_status import CoverStatus
+from elmax_api.model.panel import PanelStatus
+
+from homeassistant.components.cover import CoverEntity, CoverEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ElmaxCoordinator
+from .common import ElmaxEntity
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Elmax cover platform."""
+    coordinator: ElmaxCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    known_devices = set()
+
+    def _discover_new_devices():
+        panel_status: PanelStatus = coordinator.data
+        # In case the panel is offline, its status will be None. In that case, simply do nothing
+        if panel_status is None:
+            return
+
+        # Otherwise, add all the entities we found
+        entities = []
+        for cover in panel_status.covers:
+            # Skip already handled devices
+            if cover.endpoint_id in known_devices:
+                continue
+            entity = ElmaxCover(
+                panel=coordinator.panel_entry,
+                elmax_device=cover,
+                panel_version=panel_status.release,
+                coordinator=coordinator,
+            )
+            entities.append(entity)
+
+        if entities:
+            async_add_entities(entities)
+            known_devices.update([e.unique_id for e in entities])
+
+    # Register a listener for the discovery of new devices
+    config_entry.async_on_unload(coordinator.async_add_listener(_discover_new_devices))
+
+    # Immediately run a discovery, so we don't need to wait for the next update
+    _discover_new_devices()
+
+
+class ElmaxCover(ElmaxEntity, CoverEntity):
+    """Elmax Cover entity implementation."""
+
+    _attr_supported_features = (
+        CoverEntityFeature.OPEN | CoverEntityFeature.CLOSE | CoverEntityFeature.STOP
+    )
+
+    def __check_cover_status(self, status_to_check: CoverStatus) -> bool | None:
+        """Check if the current cover entity is in a specific state."""
+        state = self.coordinator.get_cover_state(self._device.endpoint_id).status
+        if state is None:
+            return None
```

Comment:
```suggestion
        if (state := self.coordinator.get_cover_state(self._device.endpoint_id).status) is None:
            return None
```

Reply:
Done in [3be9456](https://github.com/home-assistant/core/pull/79409/commits/3be9456aaadb07f0b007e19ed839c01f7a473271)